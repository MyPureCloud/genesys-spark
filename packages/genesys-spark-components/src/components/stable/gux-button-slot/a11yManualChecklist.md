# gux-button-slot manual accessibility testing status

**Last Updated:** Fri Dec 17 2021 16:02:58 GMT-0500 (Eastern Standard Time)
| Pass | WCAG Success Criterion | Notes |
| --- | --- | --- |
| ✅ | [2.1.1 Keyboard](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html) | - |
| ✅ | [2.1.2 No Keyboard Trap](https://www.w3.org/WAI/WCAG21/Understanding/no-keyboard-trap.html) | - |
| ✅ | [2.4.3 Focus Order](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html) | - |
| ✅ | [2.4.7 Focus Visible](https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html) | - |
| ✅ | [2.5.3 Label in Name](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html#dfn-name) | - |
| ✅ | [3.2.1 On Focus](https://www.w3.org/WAI/WCAG21/Understanding/on-focus.html) | - |
| ✅ | [3.2.2 On Input](https://www.w3.org/WAI/WCAG21/Understanding/on-input.html) | - |
| ✅ | [3.3.1 Error Identification](https://www.w3.org/WAI/WCAG21/Understanding/error-identification.html) | - |
| ✅ | [3.2.2 Labels or Instructions](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html) | - |
| ✅ | [4.1.2: Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) | - |
| ✅ | [4.1.3 Status Messages](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html) | - |
