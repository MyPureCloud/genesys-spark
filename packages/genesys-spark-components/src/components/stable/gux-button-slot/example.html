<h1>gux-button-slot</h1>

<h2>Gux Button</h2>

<div class="container">
  <gux-button onclick="notify(event)">Default</gux-button>

  <gux-button accent="primary" onclick="notify(event)">Primary</gux-button>

  <gux-button accent="secondary" onclick="notify(event)">Secondary</gux-button>

  <gux-button accent="tertiary" onclick="notify(event)">Tertiary</gux-button>

  <gux-button accent="ghost" onclick="notify(event)">Ghost</gux-button>

  <gux-button accent="danger" onclick="notify(event)">Danger</gux-button>

  <div>
    This is an
    <gux-button accent="inline" onclick="notify(event)"> Inline </gux-button>
    button.
  </div>

  <gux-button onclick="notify(event)">
    <gux-icon
      icon-name="fa/diamond-regular"
      screenreader-text="This will be read by a screen reader"
    ></gux-icon>
  </gux-button>

  <gux-button onclick="notify(event)">
    <div class="align-content">
      <gux-icon icon-name="fa/diamond-regular" decorative></gux-icon>
      <span>Text in a span element</span>
    </div>
  </gux-button>

  <gux-button accent="invalid" onclick="notify(event)">
    <div class="align-content">
      <gux-icon icon-name="fa/diamond-regular" decorative></gux-icon>
      <span>Text in a span element</span>
    </div>
  </gux-button>
</div>

<h2>Slotted Button Tag</h2>

<h3>Enabled</h3>

<div class="container">
  <gux-button-slot>
    <button type="button" onclick="notify(event)">Default</button>
  </gux-button-slot>

  <gux-button-slot accent="primary">
    <button type="button" onclick="notify(event)">Primary</button>
  </gux-button-slot>

  <gux-button-slot accent="secondary">
    <button type="button" onclick="notify(event)">Secondary</button>
  </gux-button-slot>

  <gux-button-slot accent="tertiary">
    <button type="button" onclick="notify(event)">Tertiary</button>
  </gux-button-slot>

  <gux-button-slot accent="ghost">
    <button type="button" onclick="notify(event)">Ghost</button>
  </gux-button-slot>

  <gux-button-slot accent="danger">
    <button type="button" onclick="notify(event)">Danger</button>
  </gux-button-slot>

  <div>
    This is an
    <gux-button-slot accent="inline">
      <button type="button" onclick="notify(event)">Inline</button>
    </gux-button-slot>
    button.
  </div>

  <gux-button-slot icon-only>
    <button
      type="button"
      title="This will be read by a screen reader"
      onclick="notify(event)"
    >
      <gux-icon icon-name="fa/diamond-regular" decorative></gux-icon>
    </button>
  </gux-button-slot>

  <gux-button-slot>
    <button type="button" onclick="notify(event)">
      <div class="align-content">
        <gux-icon icon-name="fa/diamond-regular" decorative></gux-icon>
        <span>Text in a span element</span>
      </div>
    </button>
  </gux-button-slot>

  <gux-button-slot accent="invalid">
    <button type="button" onclick="notify(event)">
      <div class="align-content">
        <gux-icon icon-name="fa/diamond-regular" decorative></gux-icon>
        <span>Text in a span element</span>
      </div>
    </button>
  </gux-button-slot>
</div>

<h3>Disabled</h3>

<div class="container">
  <gux-button-slot>
    <button type="button" disabled onclick="notify(event)">Default</button>
  </gux-button-slot>

  <gux-button-slot accent="primary">
    <button type="button" disabled onclick="notify(event)">Primary</button>
  </gux-button-slot>

  <gux-button-slot accent="secondary">
    <button type="button" disabled onclick="notify(event)">Secondary</button>
  </gux-button-slot>

  <gux-button-slot accent="tertiary">
    <button type="button" disabled onclick="notify(event)">Tertiary</button>
  </gux-button-slot>

  <gux-button-slot accent="ghost">
    <button type="button" disabled onclick="notify(event)">Ghost</button>
  </gux-button-slot>

  <gux-button-slot accent="danger">
    <button type="button" disabled onclick="notify(event)">Danger</button>
  </gux-button-slot>

  <div>
    This is an
    <gux-button-slot accent="inline">
      <button type="button" disabled onclick="notify(event)">Inline</button>
    </gux-button-slot>
    button.
  </div>

  <gux-button-slot icon-only>
    <button
      type="button"
      disabled
      title="This will be read by a screen reader"
      onclick="notify(event)"
    >
      <gux-icon icon-name="fa/diamond-regular" decorative></gux-icon>
    </button>
  </gux-button-slot>

  <gux-button-slot>
    <button type="button" disabled onclick="notify(event)">
      <div class="align-content">
        <gux-icon icon-name="fa/diamond-regular" decorative></gux-icon>
        <span>Text in a span element</span>
      </div>
    </button>
  </gux-button-slot>

  <gux-button-slot accent="invalid">
    <button type="button" disabled onclick="notify(event)">
      <div class="align-content">
        <gux-icon icon-name="fa/diamond-regular" decorative></gux-icon>
        <span>Text in a span element</span>
      </div>
    </button>
  </gux-button-slot>
</div>

<h2>Slotted Input Tag (button)</h2>

<div class="container">
  <gux-button-slot>
    <input type="button" onclick="notify(event)" value="Default" />
  </gux-button-slot>

  <gux-button-slot accent="primary">
    <input type="button" onclick="notify(event)" value="Primary" />
  </gux-button-slot>

  <gux-button-slot accent="secondary">
    <input type="button" onclick="notify(event)" value="Secondary" />
  </gux-button-slot>

  <gux-button-slot accent="tertiary">
    <input type="button" onclick="notify(event)" value="Tertiary" />
  </gux-button-slot>

  <gux-button-slot accent="ghost">
    <input type="button" onclick="notify(event)" value="Ghost" />
  </gux-button-slot>

  <gux-button-slot accent="danger">
    <input type="button" onclick="notify(event)" value="Danger" />
  </gux-button-slot>

  <div>
    This is an
    <gux-button-slot accent="inline">
      <input type="button" onclick="notify(event)" value="Inline" />
    </gux-button-slot>
    button.
  </div>
</div>

<h2>Slotted Input Tag (submit)</h2>

<div class="container">
  <gux-button-slot>
    <input type="submit" onclick="notify(event)" value="Default" />
  </gux-button-slot>

  <gux-button-slot accent="primary">
    <input type="submit" onclick="notify(event)" value="Primary" />
  </gux-button-slot>

  <gux-button-slot accent="secondary">
    <input type="submit" onclick="notify(event)" value="Secondary" />
  </gux-button-slot>

  <gux-button-slot accent="tertiary">
    <input type="submit" onclick="notify(event)" value="Tertiary" />
  </gux-button-slot>

  <gux-button-slot accent="ghost">
    <input type="submit" onclick="notify(event)" value="Ghost" />
  </gux-button-slot>

  <gux-button-slot accent="danger">
    <input type="submit" onclick="notify(event)" value="Danger" />
  </gux-button-slot>

  <div>
    This is an
    <gux-button-slot accent="inline">
      <input type="submit" onclick="notify(event)" value="Inline" />
    </gux-button-slot>
    button.
  </div>
</div>

<style>
  .container {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    gap: 10px;
    place-content: stretch flex-start;
  }

  .align-content {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    place-content: center center;
    align-items: center;
  }

  .align-content > * {
    flex: 0 1 auto;
    align-self: auto;
    order: 0;
  }

  gux-icon {
    width: 16px;
    height: 16px;
  }

  .align-content > gux-icon {
    padding-right: 4px;
  }
</style>
