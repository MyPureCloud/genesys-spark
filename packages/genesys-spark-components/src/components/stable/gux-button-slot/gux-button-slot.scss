@use '../../../style/focus.scss';

@mixin button-text {
  font-family: var(--gse-ui-button-text-fontFamily);
  font-size: var(--gse-ui-button-text-fontSize);
  font-weight: var(--gse-ui-button-text-fontWeight);
  line-height: var(--gse-ui-button-text-lineHeight);
}

:host {
  display: inline-block;
  pointer-events: none;
}

::slotted(input[type='button']),
::slotted(input[type='submit']),
::slotted(button) {
  @include button-text;

  width: 100%;
  min-width: var(--gse-ui-button-icon-only-width);
  height: var(--gse-ui-button-default-height);
  padding: var(--gse-ui-button-padding);
  overflow: hidden;
  color: var(--gse-ui-button-secondary-default-foregroundColor);
  text-overflow: ellipsis;
  white-space: nowrap;
  pointer-events: auto;
  cursor: pointer;
  background-color: var(--gse-ui-button-secondary-default-backgroundColor);
  border: none;
  border-radius: var(--gse-ui-button-borderRadius);
}

:host([icon-only]) {
  ::slotted(input[type='button']),
  ::slotted(input[type='submit']),
  ::slotted(button) {
    padding: var(--gse-ui-button-paddingIconOnly);
  }
}

::slotted(input[type='button'][disabled]),
::slotted(input[type='submit'][disabled]),
::slotted(button[disabled]) {
  cursor: default;
  opacity: var(--gse-ui-button-disabled-opacity);
}

::slotted(input[type='button']:focus),
::slotted(input[type='submit']:focus),
::slotted(button:focus) {
  outline: none;
}

::slotted(input[type='button']:focus-visible:enabled),
::slotted(input[type='submit']:focus-visible:enabled),
::slotted(button:focus-visible:enabled) {
  @include focus.gux-focus-ring;
}

::slotted(input[type='button']:hover:enabled),
::slotted(input[type='submit']:hover:enabled),
::slotted(button:hover:enabled) {
  color: var(--gse-ui-button-secondary-hover-foregroundColor);
  background-color: var(--gse-ui-button-secondary-hover-backgroundColor);
}

::slotted(input[type='button']:active:enabled),
::slotted(input[type='submit']:active:enabled),
::slotted(button:active:enabled) {
  color: var(--gse-ui-button-secondary-active-foregroundColor);
  background-color: var(--gse-ui-button-secondary-active-backgroundColor);
}

:host([accent='primary']) {
  ::slotted(input[type='button']),
  ::slotted(input[type='submit']),
  ::slotted(button) {
    color: var(--gse-ui-button-primary-default-foregroundColor);
    background-color: var(--gse-ui-button-primary-default-backgroundColor);
  }

  ::slotted(input[type='button']:hover:enabled),
  ::slotted(input[type='submit']:hover:enabled),
  ::slotted(button:hover:enabled) {
    color: var(--gse-ui-button-primary-hover-foregroundColor);
    background-color: var(--gse-ui-button-primary-hover-backgroundColor);
  }

  ::slotted(input[type='button']:active:enabled),
  ::slotted(input[type='submit']:active:enabled),
  ::slotted(button:active:enabled) {
    color: var(--gse-ui-button-primary-active-foregroundColor);
    background-color: var(--gse-ui-button-primary-active-backgroundColor);
  }
}

:host([accent='tertiary']) {
  ::slotted(input[type='button']),
  ::slotted(input[type='submit']),
  ::slotted(button) {
    color: var(--gse-ui-button-tertiary-default-foregroundColor);
    background-color: var(--gse-ui-button-tertiary-default-backgroundColor);
    border-color: var(--gse-ui-button-tertiary-default-border-color);
    border-style: var(--gse-ui-button-tertiary-default-border-style);
    border-width: var(--gse-ui-button-tertiary-default-border-width);
  }

  ::slotted(input[type='button']:hover:enabled),
  ::slotted(input[type='submit']:hover:enabled),
  ::slotted(button:hover:enabled) {
    color: var(--gse-ui-button-tertiary-hover-foregroundColor);
    background-color: var(--gse-ui-button-tertiary-hover-backgroundColor);
  }

  ::slotted(input[type='button']:active:enabled),
  ::slotted(input[type='submit']:active:enabled),
  ::slotted(button:active:enabled) {
    color: var(--gse-ui-button-tertiary-active-foregroundColor);
    background-color: var(--gse-ui-button-tertiary-active-backgroundColor);
  }
}

:host([accent='ghost']) {
  ::slotted(input[type='button']),
  ::slotted(input[type='submit']),
  ::slotted(button) {
    color: var(--gse-ui-button-ghost-default-foregroundColor);
    background-color: var(--gse-ui-button-ghost-default-backgroundColor);
  }

  ::slotted(input[type='button']:hover:enabled),
  ::slotted(input[type='submit']:hover:enabled),
  ::slotted(button:hover:enabled) {
    color: var(--gse-ui-button-ghost-hover-foregroundColor);
    background-color: var(--gse-ui-button-ghost-hover-backgroundColor);
  }

  ::slotted(input[type='button']:active:enabled),
  ::slotted(input[type='submit']:active:enabled),
  ::slotted(button:active:enabled) {
    color: var(--gse-ui-button-ghost-active-foregroundColor);
    background-color: var(--gse-ui-button-ghost-active-backgroundColor);
  }
}

:host([accent='danger']) {
  ::slotted(input[type='button']),
  ::slotted(input[type='submit']),
  ::slotted(button) {
    color: var(--gse-ui-button-danger-default-foregroundColor);
    background-color: var(--gse-ui-button-danger-default-backgroundColor);
  }

  ::slotted(input[type='button']:hover:enabled),
  ::slotted(input[type='submit']:hover:enabled),
  ::slotted(button:hover:enabled) {
    color: var(--gse-ui-button-danger-hover-foregroundColor);
    background-color: var(--gse-ui-button-danger-hover-backgroundColor);
  }

  ::slotted(input[type='button']:active:enabled),
  ::slotted(input[type='submit']:active:enabled),
  ::slotted(button:active:enabled) {
    color: var(--gse-ui-button-danger-active-foregroundColor);
    background-color: var(--gse-ui-button-danger-active-backgroundColor);
  }
}

:host([accent='inline']) {
  min-width: initial;

  ::slotted(input[type='button']),
  ::slotted(input[type='submit']),
  ::slotted(button) {
    height: initial;
    padding: 0;
    color: var(--gse-ui-link-default-foregroundColor);
    background: none;
    border: none;
  }

  ::slotted(input[type='button'][disabled]),
  ::slotted(input[type='submit'][disabled]),
  ::slotted(button[disabled]) {
    color: var(--gse-ui-link-disabled-foregroundColor);
  }

  ::slotted(input[type='button']:hover:enabled),
  ::slotted(input[type='submit']:hover:enabled),
  ::slotted(button:hover:enabled) {
    color: var(--gse-ui-link-hover-foregroundColor);
    text-decoration: underline;
    background: none;
  }

  ::slotted(input[type='button']:active:enabled),
  ::slotted(input[type='submit']:active:enabled),
  ::slotted(button:active:enabled) {
    color: var(--gse-ui-link-active-foregroundColor);
    text-decoration: underline;
    background: none;
  }
}
