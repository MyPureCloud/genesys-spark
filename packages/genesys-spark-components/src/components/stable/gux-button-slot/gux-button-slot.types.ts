export type GuxButtonSlotAccent =
  | 'primary'
  | 'secondary'
  | 'tertiary'
  | 'ghost'
  | 'danger'
  | 'inline'
  | 'input-icon';
