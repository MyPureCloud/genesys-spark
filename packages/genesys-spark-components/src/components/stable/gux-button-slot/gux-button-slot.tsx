import { Component, Element, h, Host, JSX, Prop } from '@stencil/core';

import { trackComponent } from '@utils/tracking/usage';
import { logError } from '@utils/error/log-error';
import { GuxButtonSlotAccent } from './gux-button-slot.types';

/**
 * @slot - button, input[type="button"] or input[type="submit"] element
 */
@Component({
  styleUrl: 'gux-button-slot.scss',
  tag: 'gux-button-slot',
  shadow: true
})
export class GuxButtonSlot {
  @Element() root: HTMLElement;

  @Prop()
  accent: GuxButtonSlotAccent = 'secondary';

  @Prop()
  iconOnly: boolean;

  private validateSlotContent(): void {
    let slottedElement = this.root.children[0];
    let slottedTagName = slottedElement.tagName;

    if (slottedTagName === 'SLOT') {
      slottedElement = (
        slottedElement as HTMLSlotElement
      ).assignedNodes()[0] as HTMLElement;
      slottedTagName = slottedElement.tagName;
    }

    if (slottedTagName === 'BUTTON') {
      return;
    } else if (slottedTagName === 'INPUT') {
      const slottedType = slottedElement.getAttribute('type');

      if (slottedType === 'button' || slottedType === 'submit') {
        return;
      }
    }

    logError(
      this.root,
      'You must slot a button, input[type="button"] or input[type="submit"] element.'
    );
  }

  componentWillLoad(): void {
    trackComponent(this.root);

    this.validateSlotContent();
  }

  render(): JSX.Element {
    return (
      <Host accent={this.accent} icon-only={this.iconOnly}>
        <slot />
      </Host>
    ) as JSX.Element;
  }
}
