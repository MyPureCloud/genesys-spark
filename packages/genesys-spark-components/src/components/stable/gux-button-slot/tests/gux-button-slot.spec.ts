jest.mock('../../../../utils/error/log-error', () => ({
  __esModule: true,
  logError: jest.fn()
}));

import { checkRenders, newSpecPage } from '@test/specTestUtils';
import { GuxButtonSlot } from '../gux-button-slot';
import { renderConfigs } from './gux-button-slot.common';

import { logError } from '../../../../utils/error/log-error';

const components = [GuxButtonSlot];
const language = 'en';

describe('gux-button-slot', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('#render', () => {
    checkRenders(renderConfigs, components);
  });

  describe('slot content', () => {
    it('should not log an error if a button element is slotted', async () => {
      await newSpecPage({
        components,
        html: '<gux-button-slot><button type="button" accent="primary">Button</button></gux-button-slot>',
        language
      });

      expect(logError).not.toHaveBeenCalled();
    });

    it('should not log an error if an input[type=button] element is slotted', async () => {
      await newSpecPage({
        components,
        html: '<gux-button-slot><input type="button" accent="primary" value="Button"/></gux-button-slot>',
        language
      });

      expect(logError).not.toHaveBeenCalled();
    });

    it('should not log an error if an input[type=submit] element is slotted', async () => {
      await newSpecPage({
        components,
        html: '<gux-button-slot><input type="submit" accent="primary" value="Button"/></gux-button-slot>',
        language
      });

      expect(logError).not.toHaveBeenCalled();
    });

    it('should not log an error if an input[type=text] element is slotted', async () => {
      await newSpecPage({
        components,
        html: '<gux-button-slot id="test"><input type="text" accent="primary" value="Button"/></gux-button-slot>',
        language
      });

      expect(logError).toHaveBeenCalledWith(
        document.getElementById('test'),
        'You must slot a button, input[type="button"] or input[type="submit"] element.'
      );
    });

    it('should not log an error if a button element is slotted', async () => {
      await newSpecPage({
        components,
        html: '<gux-button-slot id="test"><div>Not a button</div></gux-button-slot>',
        language
      });

      expect(logError).toHaveBeenCalledWith(
        document.getElementById('test'),
        'You must slot a button, input[type="button"] or input[type="submit"] element.'
      );
    });
  });
});
