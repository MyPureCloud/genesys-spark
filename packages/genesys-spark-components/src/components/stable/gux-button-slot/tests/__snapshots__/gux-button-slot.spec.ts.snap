// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`gux-button-slot #render should render button with invalid accent 1`] = `
<gux-button-slot accent="invalid">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <button type="button" accent="primary">
    Button
  </button>
</gux-button-slot>
`;

exports[`gux-button-slot #render should render danger button 1`] = `
<gux-button-slot accent="danger">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <button type="button" accent="primary">
    Button
  </button>
</gux-button-slot>
`;

exports[`gux-button-slot #render should render danger input[type=button] 1`] = `
<gux-button-slot accent="danger">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="button" accent="primary" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render danger input[type=submit] 1`] = `
<gux-button-slot accent="danger">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="submit" accent="primary" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render default button 1`] = `
<gux-button-slot accent="secondary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <button type="button" accent="primary">
    Button
  </button>
</gux-button-slot>
`;

exports[`gux-button-slot #render should render default input[type=button] 1`] = `
<gux-button-slot accent="secondary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="button" accent="primary" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render default input[type=submit] 1`] = `
<gux-button-slot accent="secondary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="submit" accent="primary" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled danger button 1`] = `
<gux-button-slot accent="danger">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <button type="button" accent="primary" disabled="">
    Button
  </button>
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled danger input[type=button] 1`] = `
<gux-button-slot accent="danger">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="button" accent="primary" disabled="" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled danger input[type=submit] 1`] = `
<gux-button-slot accent="danger">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="submit" accent="primary" disabled="" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled default button 1`] = `
<gux-button-slot accent="secondary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <button type="button" accent="primary" disabled="">
    Button
  </button>
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled default input[type=button] 1`] = `
<gux-button-slot accent="secondary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="button" accent="primary" disabled="" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled default input[type=submit] 1`] = `
<gux-button-slot accent="secondary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="submit" accent="primary" disabled="" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled ghost button 1`] = `
<gux-button-slot accent="ghost">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <button type="button" accent="primary" disabled="">
    Button
  </button>
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled ghost input[type=button] 1`] = `
<gux-button-slot accent="ghost">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="button" accent="primary" disabled="" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled ghost input[type=submit] 1`] = `
<gux-button-slot accent="ghost">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="submit" accent="primary" disabled="" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled inline button 1`] = `
<gux-button-slot accent="inline">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <button type="button" accent="primary" disabled="">
    Button
  </button>
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled inline input[type=button] 1`] = `
<gux-button-slot accent="inline">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="button" accent="primary" disabled="" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled inline input[type=submit] 1`] = `
<gux-button-slot accent="inline">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="submit" accent="primary" disabled="" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled primary button 1`] = `
<gux-button-slot accent="primary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <button type="button" accent="primary" disabled="">
    Button
  </button>
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled primary input[type=button] 1`] = `
<gux-button-slot accent="primary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="button" accent="primary" disabled="" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled primary input[type=submit] 1`] = `
<gux-button-slot accent="primary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="submit" accent="primary" disabled="" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled secondary button 1`] = `
<gux-button-slot accent="secondary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <button type="button" accent="primary" disabled="">
    Button
  </button>
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled secondary input[type=button] 1`] = `
<gux-button-slot accent="secondary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="button" accent="primary" disabled="" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled secondary input[type=submit] 1`] = `
<gux-button-slot accent="secondary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="submit" accent="primary" disabled="" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled tertiary button 1`] = `
<gux-button-slot accent="tertiary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <button type="button" accent="primary" disabled="">
    Button
  </button>
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled tertiary input[type=button] 1`] = `
<gux-button-slot accent="tertiary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="button" accent="primary" disabled="" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render disabled tertiary input[type=submit] 1`] = `
<gux-button-slot accent="tertiary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="submit" accent="primary" disabled="" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render ghost button 1`] = `
<gux-button-slot accent="ghost">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <button type="button" accent="primary">
    Button
  </button>
</gux-button-slot>
`;

exports[`gux-button-slot #render should render ghost input[type=button] 1`] = `
<gux-button-slot accent="ghost">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="button" accent="primary" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render ghost input[type=submit] 1`] = `
<gux-button-slot accent="ghost">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="submit" accent="primary" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render inline button 1`] = `
<gux-button-slot accent="inline">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <button type="button" accent="primary">
    Button
  </button>
</gux-button-slot>
`;

exports[`gux-button-slot #render should render inline button without width constraint 1`] = `
<gux-button-slot accent="inline">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <button type="button" accent="primary">
    S
  </button>
</gux-button-slot>
`;

exports[`gux-button-slot #render should render inline input[type=button] 1`] = `
<gux-button-slot accent="inline">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="button" accent="primary" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render inline input[type=submit] 1`] = `
<gux-button-slot accent="inline">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="submit" accent="primary" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render input[type=button] with invalid accent 1`] = `
<gux-button-slot accent="invalid">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="button" accent="primary" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render input[type=submit] with invalid accent 1`] = `
<gux-button-slot accent="invalid">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="submit" accent="primary" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render invalid button 1`] = `
<gux-button-slot accent="invalid">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <button type="button" accent="primary">
    Button
  </button>
</gux-button-slot>
`;

exports[`gux-button-slot #render should render invalid input[type=button] 1`] = `
<gux-button-slot accent="invalid">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="button" accent="primary" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render invalid input[type=submit] 1`] = `
<gux-button-slot accent="invalid">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="submit" accent="primary" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render primary button 1`] = `
<gux-button-slot accent="primary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <button type="button" accent="primary">
    Button
  </button>
</gux-button-slot>
`;

exports[`gux-button-slot #render should render primary input[type=button] 1`] = `
<gux-button-slot accent="primary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="button" accent="primary" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render primary input[type=submit] 1`] = `
<gux-button-slot accent="primary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="submit" accent="primary" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render secondary button 1`] = `
<gux-button-slot accent="secondary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <button type="button" accent="primary">
    Button
  </button>
</gux-button-slot>
`;

exports[`gux-button-slot #render should render secondary input[type=button] 1`] = `
<gux-button-slot accent="secondary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="button" accent="primary" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render secondary input[type=submit] 1`] = `
<gux-button-slot accent="secondary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="submit" accent="primary" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render tertiary button 1`] = `
<gux-button-slot accent="tertiary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <button type="button" accent="primary">
    Button
  </button>
</gux-button-slot>
`;

exports[`gux-button-slot #render should render tertiary input[type=button] 1`] = `
<gux-button-slot accent="tertiary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="button" accent="primary" value="Button">
</gux-button-slot>
`;

exports[`gux-button-slot #render should render tertiary input[type=submit] 1`] = `
<gux-button-slot accent="tertiary">
  <template shadowrootmode="open">
    <slot>
    </slot>
  </template>
  <input type="submit" accent="primary" value="Button">
</gux-button-slot>
`;
